lib/models/ObjectLockConfiguration.ts

```diff
@@ -89,23 +100,18 @@ export default class ObjectLockConfiguration {
      * @param dr - DefaultRetention object containing days or years
      * @return - contains error if parsing failed
      */
-    _parseTime(dr: DefaultRetention): ParsedRetention {
-        if ('Days' in dr && 'Years' in dr) {
+    private parseTime(dr: DefaultRetention): ParsedRetention {
+        if (dr.Days && dr.Years) {
             const msg = 'request xml contains both Days and Years';
             const error = errors.MalformedXML.customizeDescription(msg);
             return { error };
         }
-        const timeType = 'Days' in dr ? 'Days' : 'Years';
+        const timeType = dr.Days ? 'Days' : 'Years';
         if (!dr[timeType] || !dr[timeType][0]) {
             const msg = 'request xml does not contain Days or Years';
             const error = errors.MalformedXML.customizeDescription(msg);
             return { error };
         }
-        if (dr[timeType].length > 1) {
```

Comment:
why remove this check?

it is consistent with the array indexing we do (`dr[timeType][0]`), though I don't understand this indexing with regard to the typing: it seems to me either the type is not correct (we can an array of string, not a string), or the indexing is just taking the first digit?

Reply:
So it's indeed a string array, and it's due to how the xml parser works. The old code is correct, but the type wasn't. I'll update and add your suggestions.